podcasts/Categories/CategoryButtonStyle.swift

```diff
@@ -33,24 +34,65 @@ struct CategoryButtonStyle: ButtonStyle {
     // MARK: View
 
     let isSelected: Bool
+    let cornerStyle: CornerStyle
 
-    init(isSelected: Bool = false) {
+    enum CornerStyle {
+        case rounded
+        case circle
+
+        @available(iOS 16.0, *)
+        var shape: some Shape {
+            switch self {
+            case .rounded:
+                AnyShape(RoundedRectangle(cornerRadius: Constants.cornerRadius))
+            case .circle:
+               AnyShape(Circle())
+            }
+        }
+
+        var horizontalPadding: CGFloat {
+            switch self {
+            case .rounded:
+                Constants.Padding.roundedHorizontal
+            case .circle:
+                Constants.Padding.circleHorizontal
+            }
+        }
+    }
+
+    /// Used for generating previews with isPressed button state
+    fileprivate var forcePressed = false
+
+    init(isSelected: Bool = false, cornerStyle: CornerStyle = .rounded) {
         self.isSelected = isSelected
+        self.cornerStyle = cornerStyle
     }
 
     func makeBody(configuration: Configuration) -> some View {
         configuration.label
             .font(.callout.weight(.medium))
             .fixedSize(horizontal: true, vertical: false)
-            .padding(.horizontal, Constants.Padding.horizontal)
+            .padding(.horizontal, cornerStyle.horizontalPadding)
             .padding(.vertical, Constants.Padding.vertical)
             .cornerRadius(Constants.cornerRadius)
             .background(isSelected ? selectedBackground : (configuration.isPressed ? background : Color.clear))
             .foregroundColor(isSelected ? selectedForeground : foreground)
-            .overlay(
-                RoundedRectangle(cornerRadius: Constants.cornerRadius)
-                    .stroke(isSelected ? selectedBackground : border, lineWidth: 1)
-            )
-            .clipShape(RoundedRectangle(cornerRadius: Constants.cornerRadius))
+            .modify {
+                if #available(iOS 16.0, *) {
+                    $0
+                        .padding(cornerStyle == .circle ? 3 : 0)
+                        .overlay(
+                        cornerStyle.shape
+                            .stroke(isSelected ? selectedBackground : border, lineWidth: 1)
+                    )
+                    .clipShape(cornerStyle.shape)
+                } else {
+                    $0.overlay(
```

Comment:
I'm just falling back to RoundedRect here since `AnyShape` is iOS 16+

I don't think it's a huge deal to use the non-circular style and it didn't seem worth the effort to work around since we're likely to drop iOS 15 soonish.